### PR TITLE
Add UT for orchagent watchdog

### PR DESCRIPTION
**Pull request by liuh-80**
### Description of PR
Add UT for orchagent watchdog.

Summary:
SWSS service will add watchdog mechanism to generate keepalive message, and generate alert when swss have issue.
This PR will add new UT to cover the watchdog mechanism.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Add new UT to test and protect watchdog mechanism from code change.

#### How did you do it?
Pause orchagent service with 'kill -stop' command and check if the watchdog can send alert.

#### How did you verify/test it?
Manually test new UT.
Pass PR validation.

#### Any platform specific information?
No

#### Supported testbed topology if it's a new test case?
Any

### CODE REVIEW

File: tests/system_health/test_watchdog.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    # resume orchagent and clear syslog
+    duthost.shell(r"sudo kill -CONT {}".format(pid), module_ignore_errors=True)
+    duthost.shell(r"sudo truncate -s 0 /var/log/syslog", module_ignore_errors=True)
+
+
+def test_orchagent_watchdog(duthosts, enum_rand_one_per_hwsku_hostname, pause_orchagent):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    result = duthost.shell(
+                        r"docker exec -i swss sh -c 'test -f /usr/bin/supervisor-proc-watchdog-listener && echo exist'",
```
Comment (prgeor) [inline tests/system_health/test_watchdog.py:39]: @liuh-80 can you point me to the watchdog listner code?

Comment (liuh-80) [reply tests/system_health/test_watchdog.py:39]: @prgeor here is the listener: https://github.com/sonic-net/sonic-buildimage/pull/14686
I will update the PR later to merge the listener with process exit listener.

File: tests/system_health/test_watchdog.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    result = duthost.shell(
+                        r"docker exec -i swss sh -c 'test -f /etc/supervisor/watchdog_processes && echo exist'",
+                        module_ignore_errors=True)['stdout']
+    logger.info('Check watchdog exist: {}'.format(result))
+    if result != 'exist':
+        pytest.skip("Skip orchagent watchdog test.")
+
+    # wait watchdog emit alert
+    WATCHDOG_TIMEOUT = 120
```
Comment (qiluo-msft) [inline tests/system_health/test_watchdog.py:46]: >120 [](http://example.com/codeflow?start=23&length=3)

Magic number. How do you get the value? Is it too long or possible too short if anyone change code accidently?

File: tests/system_health/test_watchdog.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+logger = logging.getLogger(__name__)
+
+SLEEP_TIME = 10
+
+
+@pytest.fixture
+def pause_orchagent(duthost):
+    # find orchagent pid
+    pid = duthost.shell(
+                    r"ps -ef | grep orchagent | grep -v grep | awk '{print $2}'",
```
Comment (qiluo-msft) [inline tests/system_health/test_watchdog.py:20]: >ps -ef | grep orchagent | grep [](http://example.com/codeflow?start=22&length=30)

Not a blocking issue, you may consider pgrep.

File: tests/system_health/test_watchdog.py (hunk 2 above)
Comment (liuh-80) [reply tests/system_health/test_watchdog.py:46]: The watchdog will send alert after 60 seconds, so I wait for 60*2 here.

File: tests/system_health/test_watchdog.py (hunk 3 above)
Comment (liuh-80) [reply tests/system_health/test_watchdog.py]: Fixed.

**Review by qiluo-msft — APPROVED** [review]

**Comment by liuh-80** [comment]
system_health/test_watchdog.py::test_orchagent_watchdog[vlab-01] PASSED  [100%]

Will merge PR.

**Pull request merged**